lobby/src/main/java/org/triplea/lobby/server/login/LobbyLoginValidator.java

```diff
@@ -181,6 +190,11 @@ private static String getBanDurationBreakdown(final Timestamp stamp) {
           response,
           pass -> {
             if (hashedPassword.isBcrypted()) {
+              // TODO: update tests to verify tempPasswordVerification branch.
+              if (tempPasswordVerification.checkTempPassword(username, pass)) {
+                return ServerQuarantineConversation.CHANGE_PASSWORD;
+              }
+
```

Comment:
I'm not sure if I follow. If we return a null string, that will look like a success response back to the client. In essence we're returning a magic string here that the client would act on to trigger the change password prompt.

Reply:
Ah, I shoud've clarified I was reffering to the password column in the "user table", so we force a migration to the new system this way, but wouldn't actually have to use some sort of migration logic.
This would allow us to drop the column entirely and just add a non-null constraint for new values and just fill the other passwords with randomly generated ones so we don't have any null values there or something

Reply:
I think we're on the same page with forgot password being a nice feature to help with the migration.

For any user where we use a dummy password, it would be good to trigger the forgot password feature, generate a temp password and send an email. We should perhaps plan that migration though.

One point to be sure about, there is no verification of whose account you request for password reset. It would be bad to disable a password when a temp password is requested, it could be used to annoy someone and invalidate their password. That is a vector for abuse of the system, it's one reason for the audit table to store inet address so we can ban those users if we get reports about abuse.